src/http.rs

```diff
@@ -99,6 +100,42 @@ impl HttpClient {
         })
     }
 
+    /// Function `entity_timed` is like `entity` but with two optional fields `transaction_time` and `valid_time` that are of type `Option<DateTime<FixedOffset>>`.
+    pub fn entity_timed(
+        &self,
+        id: String,
+        transaction_time: Option<DateTime<FixedOffset>>,
+        valid_time: Option<DateTime<FixedOffset>>,
+    ) -> Result<Edn, CruxError> {
+        if !id.starts_with(":") {
+            return Ok(edn!({:status ":bad-request", :message "ID required", :code 400}));
```

Comment:
This returns an `Ok`? Shouldn't it be an `Err`?

Reply:
the error here is not associated with the process failing but with a bad-request to server, the `code 400` will wrap this as the correct error. Also, we cannot build a `reqwest::Error`, which is lame for this reason

Reply:
Ohh okay